File: baekjoon/단계별로 풀어보기/정렬/2108_통계학.py
# https://www.acmicpc.net/problem/2108

from sys import stdin
from collections import Counter

input = stdin.readline
li = []
n = int(input())
for _ in range(n):
    li.append(int(input()))
li.sort()
print(round(sum(li) / n))
print(li[(n - 1) // 2])
li2 = sorted(Counter(li).items(), key=lambda x: (-x[1], x[0]))
print(li[0] if n == 1 else li2[0][0] if li2[0][1] != li2[1][1] else li2[1][0])
print(max(li) - min(li))

File: C#/Baekjoon/정렬/2108_통계학.cs
//https://www.acmicpc.net/problem/2108

StreamReader sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
int n = int.Parse(sr.ReadLine());
int[] a = new int[n];
int[] b = new int[8001];
for (int i = 0; i < n; i++)
{
    int c = int.Parse(sr.ReadLine());
    a[i] = c;
    b[c + 4000]++;
}
Array.Sort(a);
Console.WriteLine((int)Math.Round(a.Average()));
Console.WriteLine(a[n / 2]);
int[] f = Array.FindAll(b, e => e == b.Max());
int m = Array.IndexOf(b, f[0]);
if (f.Length == 1) Console.WriteLine(m - 4000);
else
{
    b[m] = 0;
    Console.WriteLine(Array.IndexOf(b, f[0]) - 4000);
}
Console.WriteLine(a.Max() - a.Min());
sr.Close();